cpp/modules/luma.gl/garrow/src/array.cc

```diff
@@ -34,8 +34,15 @@ Array::~Array() {
 }
 
 void Array::setData(const std::shared_ptr<arrow::Array>& data) {
+  auto vertexFormatOptional = vertexFormatFromArrowType(data->type());
+  if (!vertexFormatOptional.has_value()) {
+    throw std::runtime_error("Unsupported data format");
+  }
+  auto vertexFormat = vertexFormatOptional.value();
+  auto vertexSize = getVertexFormatSize(vertexFormat);
+
   if (!this->_buffer || data->length() != this->_length) {
-    auto size = this->_descriptor.size() * data->length();
+    auto size = vertexSize * data->length();
     this->_buffer = this->_createBuffer(this->_device, size, wgpu::BufferUsage::CopyDst | wgpu::BufferUsage::Vertex);
```

Comment:
Why do we need this size calculation via wgpu types? 

- My assumption is that we are not actually doing anything to the array except copying it to GPU? 
- Can't we just get the length of the Array and use that as size?
- Are we trying to prevent underflows?

Reply:
Getting the size of an `array::DataType` is actually not as simple as it sounds (unless we did that mapping ourselves), mainly due to nested types. As noted above, using an `arrow` doesn't make too much sense to me for multiple reasons, we can discuss this in case I'm missing something